### PR TITLE
Add Reference Architecture and Deployment Guide

### PR DESCRIPTION
**Pull request by roooms**
This PR introduces a Reference Architecture (RA) and Deployment Guide (DG) in the same format as those found in the Vault and Consul docs. It also links to Consul docs for the Consul DG as using Consul is considered the best practice when deploying Nomad in production.

### CODE REVIEW

**Comment by Omar-Khawaja** [comment]
Overall content looks good. I will let the other reviewers comment on specifics and placement in the navigation pane. I noted a few broken links through the docs (and two bullet points that don't lead anywhere)

Broken links in the Deployment Guide:

/guides/operations/atopilot.html should be changed to /guides/operations/autopilot.html

/docs/agent/configuration/index.html should be changed to /docs/configuration/index.html

http://localhost:4567/docs/agent/options.html#_datacenter should be changed to http://localhost:4567/docs/configuration/index.html#_datacenter

http://localhost:4567/docs/agent/options.html#_data_dir should be changed to http://localhost:4567/docs/configuration/index.html#_data_dir

The following bullets in the Reference Architecture don't go anywhere:

- Network Connectivity Details
- Initialization and Configuration

**Comment by roooms** [comment]
Thanks for the review @Omar-Khawaja! Sorry about the amount of minor typos and broken links, I should have caught more of those as I ported the Deployment Guide from Consul. I've cleaned up all the ones you mentioned.

**Comment by preetapan** [comment]
@roooms @Omar-Khawaja we now have a broken link checker in the set of travis checks we run (thanks @cgbaker!) You should be able to review the results here and update the PR https://travis-ci.org/hashicorp/nomad/jobs/439738049

File: website/source/guides/operations/reference-architecture.html.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+Nomad to Consul connectivity is over HTTP and should be secured with TLS as well as a Consul token to provide encryption of all traffic. This is done using Nomad's [Automatic Clustering with Consul](/guides/operations/cluster/automatic.html).
+
+### <a name="one-dc"></a>Deployment Topology within One Datacenter
+
+A single Nomad cluster is recommended for applications deployed in the same region.
+
+Each cluster is expected to have either three or five servers. This strikes a balance between availability in the case of failure and performance, as [Raft](https://raft.github.io/) consensus gets progressively slower as more servers are added.
+
+The time taken for gossip to converge increases as more client nodes are added. Similarly, the time taken by a new server to join an existing large cluster may increase as they are replicated to the new server's log.
```
Comment (preetapan) [inline website/source/guides/operations/reference-architecture.html.md]: Nomad client nodes do not gossip with each other, they send heartbeats so this sentence should be removed.

Comment (preetapan) [inline website/source/guides/operations/reference-architecture.html.md]: suggested replace `Similarly, the time taken by a new server to join an existing large cluster may increase as they are replicated to the new server's log.`  with `A new server that joins an existing cluster takes longer to replicate data as the size of the cluster increases`

File: website/source/guides/operations/reference-architecture.html.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+~> **NOTE** For large workloads, ensure that the disks support a high number of IOPS to keep up with the rapid Raft log update rate.
+
+Nomad clients can be setup with specialized workloads as well. For example, if workloads require GPU processing, a Nomad datacenter can be created to serve those GPU specific jobs and joined to a Nomad server cluster. For more information on specialized workloads, see the documentation on [job constraints](/docs/job-specification/constraint.html) to target specific client nodes.
+
+## High Availability
+
+A Nomad server cluster is the highly-available unit of deployment within a single datacenter. A recommended approach is to deploy a three or five node Nomad server cluster. With this configuration, during a Nomad server outage, failover is handled immediately without human intervention.
+
+High-availability and data locality across datacenters requires Nomad Enterprise.
```
Comment (preetapan) [inline website/source/guides/operations/reference-architecture.html.md]: clarify

File: website/source/guides/operations/reference-architecture.html.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+---
+
+# Nomad Reference Architecture
+
+This document provides recommended practices and a reference architecture for HashiCorp Nomad production deployments. This reference architecture conveys a general architecture that should be adapted to accommodate the specific needs of each implementation.
+
+The following topics are addressed:
+
+- [Reference Architecture](#ra)
+- [Deployment Topology within One Datacenter](#one-dc)
```
Comment (rcgenova) [inline website/source/guides/operations/reference-architecture.html.md]: This would be less confusing if the sections were "Deployment Topology within a Single Region" and "Deployment Topology across Multiple Regions". Lets set the proper context in terms of Nomad's data model from the start.

File: website/source/guides/operations/reference-architecture.html.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+- [Network Connectivity Details](#net)
+- [Deployment System Requirements](#system-reqs)
+- [High Availability](#high-availability)
+- [Failure Scenarios](#failure-scenarios)
+
+This document describes deploying a Nomad cluster in combination with, or with access to, a [Consul cluster](/guides/operations/consul-integration/index.html). We recommend the use of Consul with Nomad to provide automatic clustering, service discovery, health checking and dynamic configuration.
+
+## <a name="ra"></a>Reference Architecture
+
+A Nomad cluster typically comprises three or five servers (but no more than seven) and a number of client agents. A cluster can be deployed in a single physical datacenter or it can span multiple datacenters/regions. There is an important distinction to make in that Nomad divides infrastructure into regions, which are served by one Nomad server cluster, but can manage multiple datacenters. For example, a _US Region_ can include datacenters _us-east-1_ and _us-west-2_.
```
Comment (rcgenova) [inline website/source/guides/operations/reference-architecture.html.md]: Suggest rewording: "A Nomad cluster typically comprises three or five servers (but no more than seven) and a number of client agents. Nomad differs slightly from Consul in that it divides infrastruture into regions which are served by one Nomad server cluster, but can manage multiple datacenters or availability zones. For example, a _US Region_ can include datacenters _us-east-1_ and _us-west-2_."

File: website/source/guides/operations/reference-architecture.html.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+In a Nomad multi-datacenter/multi-region architecture, communication happens via [WAN gossip](/docs/internals/gossip.html). Additionally, Nomad can integrate easily with Consul to provide features such as automatic clustering, service discovery, and dynamic configurations. Thus we recommend you use Consul in your Nomad deployment to simplify the deployment.
```
Comment (rcgenova) [inline website/source/guides/operations/reference-architecture.html.md]: Recommend changing "multi-datacenter/multi-region architecture" to just "multi-region architecture"

**Comment by roooms** [comment]
I have updated the text based on guidance from Rob. Still some final work to do on the diagrams.

**Comment by topfunky** [comment]
Do you need help with the diagrams so we can get this published?

**Comment by roooms** [comment]
@topfunky I just added the latest diagrams.

**Comment by roooms** [comment]
Please could you review the material in this PR @Omar-Khawaja and @preetapan then merge or comment? This PR introduces a Reference Architecture (RA) and Deployment Guide (DG) in the same format as those found in the Vault and Consul docs, which are proving popular on those sites.

File: website/source/guides/operations/deployment-guide.html.md [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+
+# Nomad Deployment Guide
+
+This deployment guide covers the steps required to install and configure a single HashiCorp Nomad cluster as defined in the [Nomad Reference Architecture](/guides/operations/reference-architecture.html).
+
+These instructions are for installing and configuring Nomad on Linux hosts running the systemd system and service manager.
+
+## Reference Material
+
+This deployment guide is designed to work in combination with the [Nomad Reference Architecture](/guides/operations/reference-architecture.html) and [Consul Deployment Guide](https://www.consul.io/docs/guides/deployment-guide.html). Although not a strict requirement to follow the Nomad Reference Architecture, please ensure you are familiar with the overall architecture design; for example installing Nomad server agents on multiple physical or virtual (with correct anti-affinity) hosts for high-availability.
```
Comment (Omar-Khawaja) [inline website/source/guides/operations/deployment-guide.html.md]: I would reword the last part of this paragraph to:

Although **it is** not a strict requirement to follow the Nomad Reference Architecture, please ensure you are familiar with the overall architecture design; **an example of this would be** installing Nomad server agents on multiple physical or virtual (with correct anti-affinity) hosts for high-availability.
